airbyte-integrations/connectors/source-smartsheets/source_smartsheets/source.py

```diff
@@ -0,0 +1,153 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import json
+from datetime import datetime
+from typing import Dict, Generator
+
+import smartsheet
+from airbyte_protocol import (
+    AirbyteCatalog,
+    AirbyteConnectionStatus,
+    AirbyteMessage,
+    AirbyteRecordMessage,
+    AirbyteStream,
+    ConfiguredAirbyteCatalog,
+    Status,
+    Type,
+)
+from base_python import AirbyteLogger, Source
+from smartsheet.smartsheet import Smartsheet
+
+
+# helpers
+def get_prop(col_type: str) -> Dict[str, any]:
+    props = {
+        "TEXT_NUMBER": {"type": "string"},
+        "DATE": {"type": "string", "format": "date"},
+        "DATETIME": {"type": "string", "format": "date-time"},
+    }
+    if col_type in props.keys():
+        return props[col_type]
+    else:  # assume string
+        return props["TEXT_NUMBER"]
+
+
+def get_json_schema(sheet: Smartsheet) -> Dict:
+    column_info = {i["title"]: get_prop(i["type"]) for i in sheet["columns"]}
+    json_schema = {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "type": "object",
+        "properties": column_info,
+    }
+    return json_schema
+
+
+# main class definition
+class SourceSmartsheets(Source):
+    def check(self, logger: AirbyteLogger, config: json) -> AirbyteConnectionStatus:
+        try:
+            access_token = config["access_token"]
+            spreadsheet_id = config["spreadsheet_id"] + 1
```

Comment:
why the +1?

Reply:
was an artifact from testing, trying to see what gets thrown from invalid sheet id. My bad. fixed in latest commit